tests/virt_autotest/virt_utils.pm

```diff
@@ -59,21 +59,33 @@ sub repl_repo_in_sourcefile {
     }
 }
 
-sub repl_guest_autoyast_addon_with_daily_build_module {
-    #replace the addons url in guest autoyast file in qa_lib_virtauto-data with the daily build module repos
+sub repl_addon_with_daily_build_module_in_files {
+    my $file_list = shift;
+
+    $file_list =~ s/\n/ /g;
     my $version = get_version_for_daily_build_guest;
-    $version =~ s/-/\//;
-    my $autoyast_root_dir = "/usr/share/qa/virtautolib/data/autoinstallation/sles/" . $version . "/";
+    $version =~ s/-fcs//;
+    $version = uc($version);
     my $command
-      = "for file in \`find $autoyast_root_dir -type f\`;do "
-      . "sed -ri 's#^.*(Basesystem|Desktop-Applications|Legacy|Server-Applications|Development-Tools).*\$#"
-      . "<media_url>http://openqa.suse.de/assets/repo/SLE-15-Module-\\1-POOL-x86_64-Build"
+      = "for file in $file_list;do "
+      . "sed -ri 's#^.*(Basesystem|Desktop-Applications|Legacy|Server-Applications|Development-Tools|Web-Scripting).*\$#"
+      . "<media_url>http://openqa.suse.de/assets/repo/SLE-${version}-Module-\\1-POOL-x86_64-Build"
       . get_required_var('BUILD')
       . "-Media1/</media_url>#' \$file;done";
     assert_script_run($command);
     save_screenshot;
-    assert_script_run("grep media_url $autoyast_root_dir/* -r");
+    assert_script_run("grep media_url $file_list -r");
     save_screenshot;
+
+}
+
+sub repl_guest_autoyast_addon_with_daily_build_module {
+    #replace the addons url in guest autoyast file in qa_lib_virtauto-data with the daily build module repos
+    my $version = get_version_for_daily_build_guest;
+    $version =~ s/-/\//;
+    my $autoyast_root_dir = "/usr/share/qa/virtautolib/data/autoinstallation/sles/" . $version . "/";
+    my $file_list         = &script_output("find $autoyast_root_dir -type f");
```

Comment:
how about this `find $autoyast_root_dir -type f | xargs` to treat multi-line output as a single line as well as to remove '\n', then no extra `s/\n/ /`in `repl_addon_with_daily_build_module_in_files`

Reply:
Well, my design seems better. repl_addon_with_daily_build_module_in_files is a common function, which accepts files splitted by either new line or space. Such code seems clearer.

Reply:
Ok, that's fine :)